src/magnet_uri.cpp

```diff
@@ -258,14 +258,68 @@ namespace libtorrent {
 			return p;
 		}
 
+		auto select_pos = std::string::npos;
+		string_view select = url_has_argument(uri, "so", &select_pos);
+		while (!select.empty())
+		{
+			// parse the ranges or indices
+			do
+			{
+				string_view token;
+				std::tie(token, select) = split_string(select, ',');
+
+				int idx1, idx2;
+				// TODO: what's the right number here?
+				constexpr int max_index = 10000; // can't risk out of memory
+
+				auto const divider = token.find_first_of('-');
```

Comment:
I would imagine ``split_string()`` is an even better fit here, since you only expect there to be a single "-", so you get both sides back right away.

Reply:
to know whether it was found or not, see if the second result is an empty string

Reply:
OK

Reply:
looking at this again, I need the index to know if the `-` is at the beginning or right at the end